### DISCUSSION TITLE
PerfView GC Stats empty for .Net Core services

### DISCUSSION THREAD

**Discussion by Mies75** [microsoft/perfview#1503] [Q&A] [answered]
Hi!

I am investigating high memory consumption for our dotnet core 2.1 services.

The setup:
- ASP.NET Core 2.1 services
- PerfView 2.0.72, with default settings (15th of Sept 2021)
- Either IIS on a Windows 2016 server, or VS2019 / IIS express on Windows 10.
- SoapUI LoadTest hammering the REST methods

After collecting for about 20 seconds I open the GCStats window in PerfView, since I am interested in decimating Gen2 garbage collection and GC pauses. For every .NET Core service all of the GC Stats are 0 or NaN, including all data in the GC Rollup table. When I navigate to our .NET Framework services I see that all figures ARE filled.

What do I have to do to make GC Stats available in PerfView for .NET Core code?

Side note: Peak Virtual Memory Usage shows 2.223.122,727 MB for a .NET Core service, isn't that unreasonably high? Peak Process Working Set only displays 231,641 MB.

![perfview](https://user-images.githubusercontent.com/27344755/135083108-238dfd92-a388-46d3-ad62-78765aeb7334.jpg)

**Comment by brianrob** [comment, marked as answer]
@Mies75, can you share the PerfView collection command that you used?  The default collection settings include GC events, and so they should already be present.  Do you see GCs for other processes?  Also, was there load applied to this process at the time you were collecting?

**Reply by Mies75** [reply]
@brianrob, damn, I think you're right. While the process is running, at that time there are hardly any CPU Msec.
My bad, I forgot to check that! This can be closed.
